core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlPrepareHandlerBase.java

```diff
@@ -284,6 +287,8 @@ private DefaultPreparedStatement cache(DefaultPreparedStatement preparedStatemen
       LOG.warn(
           "Re-preparing already prepared query. "
               + "This is generally an anti-pattern and will likely affect performance. "
+              + "The cached version of the PreparedStatement will be returned which may have a "
+              + "different base Statement configuration than the base Statement that was prepared. "
```

Comment:
That's a bit cryptic for the uninitiated, it's hard to explain in such a short message. How about:

> The cached version of the PreparedStatement will be returned, which may use different bound statement execution parameters (CL, timeout, etc.) from the current session.prepare call.